frontend/lib/models/test_results_filters.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2023 Canonical Ltd.
+//
+// This file is part of Test Observer Frontend.
+//
+// Test Observer Frontend is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3, as
+// published by the Free Software Foundation.
+//
+// Test Observer Frontend is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'package:dartx/dartx.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+import 'execution_metadata.dart';
+import '../routing.dart';
+
+part 'test_results_filters.freezed.dart';
+part 'test_results_filters.g.dart';
+
+@freezed
+abstract class TestResultsFilters with _$TestResultsFilters {
```

Comment:
This new model is what houses all the test results filters. This matches the backend implementation created in https://github.com/canonical/test_observer/pull/389.

This is useful because now it is now all contained in this class.